---
title: 'Analysis of SARS-CoV-2 variants in Zimbabwe'
author: "Audrey Koti"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview

Zimbabwe recorded its first COVID-19 case in March 2020 and by the end of the year, sequencing results showed an increase in the occurrence of variants in the SARS-CoV-2 genome found in samples collected nationwide (MashemedRxiv). This influx of variants was likely caused by increased travel between South Africa, where the B.1.351 variant (MasheLancet) was first reported, and Zimbabwe during the December holiday season. Although the government issued social-distancing and lockdown measures, these measures were not followed or enforced strictly. This may be why several variants spread among the population as shown by the data from the surveillance carried out between the end of 2020 and January 2021. This surveillance of the SARS-CoV-2 genome also identified the C.2 and A.23.1 variants in addition to the B.1.351 variant. However, the occurrence of the C.2 and A.23.1 variants declined in Zimbabwe at the end of the surveillance in January 2021 (MasheLancet). All three SARS-CoV-2 variants are associated with increased transmission. Moreover, the B.1.351 variant is also associated with higher mortality as observed in South Africa. This report was an analysis of the occurrence of mainly single nucleotide polymorphisms (SNPs) in the SARS-CoV-2 genes in people of different ages and sex. Most COVID-19 patients from Zimbabwe were male and more than half of the cases reported occurred among people of ages 20 - 40 (MashemedRxiv). This may be why Zimbabwe's deaths were only about 2.9% of the deaths in neighboring South Africa (OxCOVID19), despite Zimbabwe having little enforcement of the guidelines to curb transmission, as people in the age group of 20 - 40 years have a better chance of recovering from COVID-19 than those older. Variant analysis is important in understanding how SARS-CoV-2 mutates and how that affects public health measures. I analyzed the distinct SNPs in SARS-CoV-2 genes grouped by age and for each sex. Next, I looked at the unique alternative alleles at each position of the sequence sorted by age and by sex. Finally, I counted the alternative alleles in the gene for the spike protein to determine the most common allele that may be the most responsible for increased transmission. I found that there is a strong association between the "T" alternative allele and males aged 30 - 50, which may have led to the higher number of COVID-19 cases among this group of people in Zimbabwe.

# Methods

This report looked at the dominant variants of the SARS-CoV-2 genes that are of concern in Zimbabwe. I obtained the data from the National Center for Biotechnology Information (ZimData). I carried out the analysis of these variants in `RStudio`[@RStudio] using R functions and packages such as `ggplot2`[@ggplot2] to plot figures, and `dplyr`[@dplyr] to group by variables as well as count up observations from the data.

## Analysis of COVID-19 cases in Zimbabwe

I used `ggplot2`[@ggplot2] to represent, on a bar plot filled by the host sex, the number of COVID-19 cases in male and female patients 18 years and older. This was done to better understand which age group and sex would have better data for variant analysis.

## Analysis of Distinct SNPs in SARS-CoV-2 Genes

Using a script written by Professor Zimmerman, I looked at the number of distinct single nucleotide polymorphisms within each named gene to determine where the majority of the variants were found. The number for each gene was tallied up after grouping by the position of the variations in the sequence. This involved using functions such as `filter()`, `tally()` and `group_by()` from the `dplyr`[@dplyr] package. I then used the a scatter plot to visualize the occurrence of SARS-CoV-2 genes in the two sexes to confirm whether or not the genes with more unique SNPs were the same for each sex. After determining the gene with the most unique SNPs, I counted the variations in that particular gene to understand the spread of these variations in males and females of different ages. I then created a line graph showing when the variations increased in Zimbabwe in the time the surveillance was carried out. I used the `lubridate` package[@lubridate] to load the collection date in the correct format.

## Analysis of Distinct SNPs in SARS-CoV-2 Genes by Sex and Age

I used a table made by Professor Zimmerman to look at the sizes of each gene. Based on the longest gene with the most unique SNPs, I analyzed the most common unique SNP in the different patient sexes and ages to determine how the occurrence of this SNP was distributed. I created a table that counts the number of times the alternative alleles occur in the gene and plotted a bar plot for the distribution of the most commonly occurring allele in male and female patients aged 18 and older using `gglopt2`[@ggplot2]. Lastly, I created a table to show whether this same alternative allele was also the most commonly occurring in patients under the age of 18, who represent the minority of COVID-19 cases in Zimbabwe.

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")
library("lubridate")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

# Figures

```{r demographics-plot}
# create a plot that represents the number of cases by age and sex
vcf_with_metadata %>%
  group_by(Host_Age, host_sex) %>%
  tally() %>%
  filter(Host_Age > 18) %>%
  ggplot(aes(x = Host_Age,
             y = n,
             fill = host_sex)) +
  geom_col(position = position_dodge()) +
  scale_fill_manual(values = c("orange", "darkgreen")) +
  labs(title = "Zimbabwe COVID-19 Patient Demographics",
       x = "Age") +
  theme_few()
```

**Figure 1**: Figure 1 represents the demographics of COVID-19 patients
# mostly men infected (more interactions at work). ages 25 - 65 but most men

```{r distinct-snps-plot}
# create a plot of unique SNP locations within each gene across all samples
vcf_with_metadata %>%
  filter(!is.na(gene)) %>% # get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>% # this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>% # this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
  geom_col(fill = "lightblue") +
  labs(title = "Count of Distinct SNPs in Named SARS-CoV-2 Genes",
         x = "Gene Name") +
  theme_few() # get rid of the gray background
```

**Figure 2**: Figure 1 shows the locations of unique SNPs for each SARS-CoV-2 gene.
# N and S genes have more unique SNPs in the set of samples analyzed.

```{r distinct-snps-by-sex-plot}
# plot the distinct SNPs on the SARS-CoV-2 genes by sex
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(gene, pos, host_sex) %>%
  tally() %>%
  group_by(gene, host_sex) %>%
  tally() %>%
   ggplot(aes(x = gene,
             y = n,
             color = host_sex)) +
  geom_point() +
  scale_color_manual(values = c("yellow", "blue")) +
  labs(title = "Occurrence of Distinct SARS-CoV-2 gene SNPs by Sex",
       x = "Gene Name",
       y = "Count") +
  theme_few()
```

**Figure 3**: Figure 3 shows the distribution of SNPs in each sex.
# men have more variations in most genes, in particular the N and S genes

```{r s-gene-snps-plot}
# plot the occurrence of variation in the S gene by age and sex
vcf_with_metadata %>%
  filter(Host_Age >= 18, !is.na(gene)) %>%
  group_by(gene, pos, Host_Age, host_sex) %>%
  tally() %>%
  group_by(gene, Host_Age, host_sex) %>%
  tally() %>%
  filter(gene == "S") %>%
  ggplot(aes(x = Host_Age,
             y = n,
             fill = host_sex)) +
  geom_col(position = position_dodge()) +
  labs(title = "Occurrence of the S Gene Variation by Age and Sex",
       x = "Age",
       y = "Count") +
  theme_few()
```

**Figure 4**: Figure 4 shows the count of the S gene variations in patients older than 18 years.
# S variations more common in males ages 30 -65, also women ages ~35 (increased interactions at work)

```{r snp-dates-plot}
# plot the occurrence of the S gene variation by date of sample collection
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(gene, pos, Collection_Date) %>%
  tally() %>%
  group_by(gene, Collection_Date) %>%
  tally() %>%
  filter(gene == "S") %>%
  ggplot(aes(x = ymd(Collection_Date),
             y = n)) +
  geom_line(color = "brown") +
  labs(title = "Occurrence of the S Gene Variation by Date",
       x = "Date",
       y = "Count") +
  theme_few()
```

**Figure 5**: Figure 5 shows the occurrence of S gene variants between December 2020 and January 2021.
# spikes around late Dec 2020 to early Jan 2021 (holiday season)

```{r alt-allele-plot}
# create a plot to show how the most common alternative allele is spread
vcf_with_metadata %>%
  filter(Host_Age >= 25 & Host_Age <= 65, !is.na(gene), alt == "T") %>%
  group_by(gene, pos, alt, Host_Age, host_sex) %>%
  tally() %>%
  group_by(gene, alt, Host_Age, host_sex) %>%
  tally() %>%
  filter(gene == "S") %>%
  ggplot(aes(x = Host_Age,
             y = n,
             fill = host_sex)) +
  geom_col(position = position_dodge()) +
  labs(title = "Distribution of the T Allele by Age and Sex",
       x = "Age",
       y = "Count") +
  theme_few() 
```

**Figure 6**: Figure 6 is based on Table 2 and shows the sex and age where the most common S gene alternative allele is most common.
# most common in males

# Tables

```{r gene-length-table}
# A table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

**Table 1**: Table 1 shows the length of each SARS-CoV-2 gene.
#Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

```{r s-gene-variants-table}
# create a table that shows how many times each alternative allele occurs in
# the spike protein gene in all patients
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(gene, pos, alt) %>%
  tally() %>%
  group_by(gene, alt) %>%
  tally() %>%
  filter(gene == "S") %>%
  knitr::kable(col.names = c("Gene Name",
                             "Alternative Allele",
                             "Count"))
```

**Table 2**: Table 2 shows the count of alternative alleles found on the S gene.
# T is the most common allele, more than 4 times the next common alleles. most common are single nucleotide polymorphisms

```{r s-gene-variants-by-age-table}
# create a table that shows how many times each alternative allele occurs in
# the spike protein gene in patients under 18
vcf_with_metadata %>%
  filter(Host_Age < 18, !is.na(gene)) %>%
  group_by(gene, pos, alt) %>%
  tally() %>%
  group_by(gene, alt) %>%
  tally() %>%
  filter(gene == "S") %>%
  knitr::kable(col.names = c("Gene Name",
                             "Alternative Allele",
                             "Count"))
```

**Table 3**: Table 3 shows the count of alternative alleles in the S gene in patients under 18.
# the T allele is still most common, two alternative alleles present in rest of population not found here (the TT and TTTA)

# Sources Cited
