---
title: 'Variant Analysis of SARS-CoV-2 in Zimbabwe between December 2020 and January 2021'
author: "Audrey Koti"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview

Zimbabwe recorded its first COVID-19 case in March 2020 and by the end of the year, sequencing results showed an increase in the occurrence of variants in the SARS-CoV-2 genome found in samples collected nationwide [@MashemedRxiv]. This influx of variants was likely caused by increased travel between South Africa, where the B.1.351 variant [@MasheLancet] was first reported, and Zimbabwe during the December holiday season. A surveillance of the SARS-CoV-2 genome also identified the C.2 and A.23.1 variants in Zimbabwe between December 2020 and January 2021 [@MasheLancet]. The three SARS-CoV-2 variants are associated with increased transmission. Moreover, the B.1.351 variant is also associated with higher mortality as observed in South Africa. However, the occurrence of the C.2 and A.23.1 variants declined in Zimbabwe at the end of the surveillance in January 2021. This report was an analysis of the occurrence of mainly single nucleotide polymorphisms (SNPs) in the SARS-CoV-2 genes in people of different ages and sex. Most COVID-19 patients from Zimbabwe were male and more than half of the cases reported occurred among people of ages 20 - 40 [@MashemedRxiv]. This may be why Zimbabwe's deaths were only about 2.9% of the deaths in neighboring South Africa [@OxCOVID19], despite Zimbabwe having little enforcement of the guidelines to curb transmission, as people in the age group of 20 - 40 years have a better chance of recovering from COVID-19 than those older. Variant analysis is important in understanding how SARS-CoV-2 mutates and how that affects public health measures. I analyzed the distinct SNPs in SARS-CoV-2 genes grouped by age and for each sex. Next, I looked at the unique alternative alleles at each position of the sequence sorted by age and by sex. Finally, I counted the alternative alleles in the gene for the spike protein to determine the most common allele that may be the most responsible for transmission. >>><<<I found that there was a strong association between the >>><<< alternative allele and this age group and sex, which may have led to the higher number of COVID-19 cases among this group of people in Zimbabwe.>>><<<

# Methods

This report looked at the dominant variants of the SARS-CoV-2 genes that are of concern in Zimbabwe. I obtained the data from the National Center for Biotechnology Information [@ZimData]. I carried out the analysis of these variants in `RStudio`[@RStudio] using R functions and packages such as `ggplot2`[@ggplot2] to plot figures, and `dplyr`[@dplyr] to group by variables as well as count up observations from the data.

## Analysis of Distinct SNPs in SARS-CoV-2 Genes

Using a script written by Professor Naupaka Zimmerman, I looked at the number of distinct single nucleotide polymorphisms within each named gene to determine where the majority of the variants were found. The number for each gene was tallied up after grouping by the position of the variations in the sequence. This involved using functions such as `filter()`, `tally()` and `group_by()` from the `dplyr`[@dplyr] package.

## Distinct SNPs in SARS-CoV-2 Genes by Age and Sex

I used the `facet_wrap` function to visualize the occurrence of unique alternative alleles at each position by age and by sex using `gglopt2`[@ggplot2]. I also counted up these alternative alleles to find out which one is the most commonly occurring allele among different ages and sexes and represented the findings in a table. 

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

# Figures

```{r distinct-snps-plot}
# create a plot of unique SNP locations within each gene across all samples
vcf_with_metadata %>%
  filter(!is.na(gene)) %>% # get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>% # this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>% # this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
    geom_col() +
    labs(title = "Count of distinct SNPs in Named SARS-CoV-2 Genes",
         x = "Gene Name") +
  theme_few() # get rid of the grey background
```

**Figure 1**: N and S genes have more unique SNPs in the set of samples analyzed.

```{r distinct-snps-by-age-plot}
# facet wrap the distinct SNPs on the SARS-CoV-2 genes by age
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(gene, pos) %>%
  tally() %>%
  group_by(gene) %>%
  tally() %>%
  ggplot(aes(x = gene,
             y = n)) +
  geom_point(size = 3, alpha = 0.3) +
  facet_wrap(. ~ age) +
  labs(title = "Occurrence of Distinct SARS-CoV-2 gene SNPs by Age",
       x = "Gene Name",
       y = "Count") +
  theme_few()
```

**Figure 2**: Figure 2 shows which SNPs are commonly found in different patient ages.

```{r distinct-snps-by-sex-plot}
# facet wrap the distinct SNPs on the SARS-CoV-2 genes by sex
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(gene, pos) %>%
  tally() %>%
  group_by(gene) %>%
  tally() %>%
  ggplot(aes(x = gene,
             y = n)) +
  geom_point(size = 3, alpha = 0.3) +
  facet_wrap(. ~ sex) +
  labs(title = "Occurrence of Distinct SARS-CoV-2 gene SNPs by Sex",
       x = "Gene Name",
       y = "Count") +
  theme_few()
```

**Figure 3**: Figure 3 shows which SNPs are commonly found in each sex.

# Tables

```{r gene-length-table}
# A table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

**Table 1**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

```{r S-gene-variants-table}
# create a table that shows how many times each alternative allele occurs in the spike protein gene

```

**Table 2**: >>><<<

# Sources Cited
